File: LivrusVirtus/Kylix3/pascal/fontes/ConPaises.pas
unit ConPaises;

interface

uses
  SysUtils, Types, Classes, Variants, QTypes, QGraphics, QControls, QForms, 
  QDialogs, QStdCtrls, Consultas, FMTBcd, Provider, DBClient, DB, SqlExpr,
  QGrids, QDBGrids, QExtCtrls;

type
  TfrmConPaises = class(TfrmConsultas)
    procedure FormClose(Sender: TObject; var Action: TCloseAction);
    procedure btnLimparClick(Sender: TObject);
    procedure btnPesquisarClick(Sender: TObject);
    procedure FormShow(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  frmConPaises: TfrmConPaises;

implementation

uses Principal, ConsultasSQL;

{$R *.xfm}

procedure TfrmConPaises.FormClose(Sender: TObject;
  var Action: TCloseAction);
begin
frmPrincipal.Paises2.Enabled:=true;
inherited;
end;

procedure TfrmConPaises.btnLimparClick(Sender: TObject);
begin
  inherited;
LimparDados(ConsSQL.Pais('S','0',''));
end;

procedure TfrmConPaises.btnPesquisarClick(Sender: TObject);
begin
  inherited;
PesquisarDados(ConsSQL.Pais('S','', edtDescricao.Text),
'País não encontrado!');
end;

procedure TfrmConPaises.FormShow(Sender: TObject);
begin
  inherited;
btnLimparClick(self);
end;

end.

File: LivrusVirtus/Kylix3/cpp/fontes/ConPaises.cpp
//---------------------------------------------------------------------------

#include <clx.h>
#pragma hdrstop

#include "ConPaises.h"
#include "ConsultasSQL.h"
#include "Principal.h"
//---------------------------------------------------------------------------
#pragma link "Consultas"
#pragma resource "*.xfm"
TfrmConPaises *frmConPaises;
//---------------------------------------------------------------------------
__fastcall TfrmConPaises::TfrmConPaises(TComponent* Owner)
        : TfrmConsultas(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TfrmConPaises::FormClose(TObject *Sender,
      TCloseAction &Action)
{
frmPrincipal->Paises2->Enabled=true;
TfrmConsultas::FormClose(Sender, Action);
}
//---------------------------------------------------------------------------
void __fastcall TfrmConPaises::btnLimparClick(TObject *Sender)
{
LimparDados(ConsSQL->Pais('S',"0",""));
}
//---------------------------------------------------------------------------
void __fastcall TfrmConPaises::btnPesquisarClick(TObject *Sender)
{
PesquisarDados(ConsSQL->Pais('S',"", edtDescricao->Text.c_bstr()),
"País não encontrado!");
}
//---------------------------------------------------------------------------
void __fastcall TfrmConPaises::FormShow(TObject *Sender)
{
btnLimparClick(this);
}
//---------------------------------------------------------------------------
